docs/concepts/workers.md

```diff
@@ -4,4 +4,165 @@ title: Workers
 sidebar_label: Workers
 ---
 
-A Worker is a service that hosts the [Workflow](/docs/concepts/workflows) and [Activity](/docs/concepts/activities) implementations.
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+A Worker is a service that:
+
+- hosts (registers) [Workflows](/docs/concepts/workflows) and [Activities](/docs/concepts/activities)
+- listens to [Activity Task Queues](/docs/concepts/task-queues) by long polling
+
+Workers should have access to the resources needed to perform the activities they implement, for example:
+
+- network access for external API calls
+- credentials for infrastructure provisioning
+- specialized GPUs for machine learning tasks
+
+Since workers are subscribed to task queues they can implement, it is straightforward to do [Activity Task Routing](/docs/concepts/activities/#activity-task-routing-through-task-queues). 
+
+> Note: if you need to process stateful work sequentially on the same machine, the Go SDK also offers a [Sessions API](https://docs.temporal.io/docs/go/sessions/).
+
+## Example Worker Code
+
+Just for illustration purposes:
```

Comment:
i'm not REALLY sure about this one, but i thought it'd be nice - let me know what you think! happy to remove.

Reply:
```suggestion
## Example Worker Code

Here is some sample code to help you understand what a Worker does (as described above). Please refer to [each SDK's documentation](/docs/application-development) for a fuller description.
```